connectors/src/connectors/google_drive/index.ts

```diff
@@ -561,6 +561,46 @@ export async function retrieveGoogleDriveObjectsTitles(
   return new Ok(titles);
 }
 
+export async function retrieveGoogleDriveContentNodes(
+  connectorId: ModelId,
+  internalIds: string[]
+): Promise<Result<ConnectorNode[], Error>> {
+  const folderOrFiles = await GoogleDriveFiles.findAll({
+    where: {
+      connectorId: connectorId,
+      driveFileId: internalIds,
+    },
+  });
+
+  const nodes = await concurrentExecutor(
+    folderOrFiles,
+    async (f): Promise<ConnectorNode> => {
+      return {
+        provider: "google_drive",
+        internalId: f.driveFileId,
+        parentInternalId: null,
+        type: getPermissionViewType(f),
+        title: f.name || "",
+        dustDocumentId: getGoogleDriveEntityDocumentId(f),
+        lastUpdatedAt: f.lastUpsertedTs?.getTime() || null,
+        sourceUrl: null,
```

Comment:
Why is source URL empty?

Reply:
Because it's empty when we retrieve the read permissions and out of consistency I kept the exact same behavior!

Reply:
Actually I think it's better to have it and the consistency should be with all connector on this screen instead of per connector. I added something, looks like it's working but let me know if you think it's better to fetch the API to retrieve the url from the API